bench/abseil-cpp/optimized/numbers.cc.ll

```diff
@@ -1134,18 +1134,18 @@ if.then98.i:                                      ; preds = %if.then92.i
 
 if.else105.i:                                     ; preds = %if.then92.i
   %5 = tail call range(i64 0, 65) i64 @llvm.ctlz.i64(i64 %conv101.i, i1 true)
-  %shl.i.i = shl i64 %conv101.i, %5
+  %6 = tail call i64 @llvm.fshl.i64(i64 %conv101.i, i64 0, i64 %5)
```

Comment:
Regression

Reply:
Missing fold: https://alive2.llvm.org/ce/z/3oTEop

Reply:
Reported: https://github.com/llvm/llvm-project/issues/122235

Reply:
Ah sorry, I wrote this issue before I saw your comment. Masking the shift amount instead of making it a pre-condition may be better...